examples/kubectl.rs

```diff
@@ -163,14 +163,15 @@ impl App {
             std::fs::read_to_string(&pth).with_context(|| format!("Failed to read {}", pth.display()))?;
         for doc in multidoc_deserialize(&yaml)? {
             let obj: DynamicObject = serde_yaml::from_value(doc)?;
+            let namespace = obj.metadata.namespace.clone().or_else(|| self.namespace.clone());
```

Comment:
Does this even need to be cloned?

I cant test right now but it looks like something like ` obj.metadata.namespace.as_ref().unwrap_or(&self.namespace)` should work.

Reply:
I also tried with ```as_ref()```, and not frequently used ```clone```,
but focus on ```namespace``` data type ,mapping within ```apply``` action, ```obj.metadata.namespace``` is ```Option<String>```  type and ```self.namespace``` also is same.

if you used ```.as_ref()``` with ```obj.metadata.namespace```, it will be ```Option<&String>``` Type, and ```unwrap_or(&self.namespace)``` with success will got ```&String```, and error will got ```&Option<String>``` from ```&self.namespace```, its type will be difference in ```namespace``` variable .

Furthermore the type of ```&Option<String>```  on ```namespace``` Field is needed within ```self::apply```.

All I know, here ```Option<String>``` , ```Option<&String>```, ```&Option<String>``` is three of difference type in Rust.


<img width="1403" alt="截圖 2023-05-19 下午5 22 59" src="https://github.com/kube-rs/kube/assets/27864573/c2274907-0308-410d-997f-8da2e1dd957b">

Reply:
makes sense. we don't need to over-optimize in the examples here anyway, just want something that's easy for users to replicate, so this is fine :-)

Reply:
No, you would use `as_deref` and `dynamic_api` should use `Option<&str>`, the current `&Option<String>` is somewhat of a code smell. The entire diff looks like this:

```diff
diff --git a/examples/kubectl.rs b/examples/kubectl.rs
index 212ccaf..3e69d7b 100644
--- a/examples/kubectl.rs
+++ b/examples/kubectl.rs
@@ -163,7 +163,7 @@ impl App {
             std::fs::read_to_string(&pth).with_context(|| format!("Failed to read {}", pth.display()))?;
         for doc in multidoc_deserialize(&yaml)? {
             let obj: DynamicObject = serde_yaml::from_value(doc)?;
-            let namespace = obj.metadata.namespace.clone().or_else(|| self.namespace.clone());
+            let namespace = obj.metadata.namespace.as_deref().or(self.namespace.as_deref());
             let gvk = if let Some(tm) = &obj.types {
                 GroupVersionKind::try_from(tm)?
             } else {
@@ -171,7 +171,7 @@ impl App {
             };
             let name = obj.name_any();
             if let Some((ar, caps)) = discovery.resolve_gvk(&gvk) {
-                let api = dynamic_api(ar, caps, client.clone(), &namespace, false);
+                let api = dynamic_api(ar, caps, client.clone(), namespace, false);
                 trace!("Applying {}: \n{}", gvk.kind, serde_yaml::to_string(&obj)?);
                 let data: serde_json::Value = serde_json::to_value(&obj)?;
                 let _r = api.patch(&name, &ssapply, &Patch::Apply(data)).await?;
@@ -208,7 +208,7 @@ async fn main() -> Result<()> {
             wc = wc.labels(label);
         }

-        let api = dynamic_api(ar, caps, client, &app.namespace, app.all);
+        let api = dynamic_api(ar, caps, client, app.namespace.as_deref(), app.all);

         tracing::info!(?app.verb, ?resource, name = ?app.name.clone().unwrap_or_default(), "requested objects");
         match app.verb {
@@ -228,7 +228,7 @@ fn dynamic_api(
     ar: ApiResource,
     caps: ApiCapabilities,
     client: Client,
-    ns: &Option<String>,
+    ns: Option<&str>,
     all: bool,
 ) -> Api<DynamicObject> {
     if caps.scope == Scope::Cluster || all {
```

Reply:
ah, yeah, this also makes sense sorry, may have merged a bit quickly.
anyone that wants to update it, will get a quick merge also